### DISCUSSION TITLE
Is this working?

### DISCUSSION THREAD

**Discussion by vhong8820** [nxvvvv/safe-exam-browser-bypass#10] [Q&A] [answered]
I successfully opened up the test start screen through VM but the test is not opened yet. Does SEB do another VM check other than when it launches?

**Comment by nxvvvv** [comment]
> I successfully opened up the test start screen through VM but the test is not opened yet. Does SEB do another VM check other than when it launches?

I'm not sure about v3.6 of SEB but the versions before that, I've modified such a way that it will work in the VM

**Reply by vhong8820** [reply]
I am currently using version 3.5, and the Safe Exam Browser (SEB) launches without issues. Which I'm going to guess its working. However, I am interested to know whether SEB has a feature that reports any irregularities or anomalies to the school during a test because I couldn't find anything helpful on their website. Could you provide information on how SEB functions in this context?

**Reply by nxvvvv** [reply]
Well, in the context of Safe Exam Browser (SEB), the detection of anomalies and other security measures are typically initiated by specific codes within the software. What I've been working on is essentially filling those specific detection logics with dummy data. This approach ensures that the SEB won't be able to detect any irregularities, and it will simply print out logs as if everything is normal.

It's important to note that the detection process occurs only once, as SEB believes the system is fully locked down by the exam environment. Generally, SEB logs are not transmitted to the school or university conducting the exam, as outlined in their [privacy statement](https://safeexambrowser.org/about_overview_en.html#privacy-statement:~:text=Log%20files%20are%20not%20transmitted%20to%20any%20centralized%20server%20by%20SEB). However, if the institution is running its own SEB server, they do have the capability to automatically retrieve the logs, as mentioned [here](https://safeexambrowser.org/about_overview_en.html#privacy-statement:~:text=will%20be%20transferred%20to%20the%20individual%20SEB%20Server%20instance).

I've also made adjustments to the logging system so that it prints only the information that would typically be logged when nothing suspicious is detected, rather than exposing any details related to cheat detection. This ensures a more discreet operation during exams. If you have any further questions or need additional information, feel free to ask!

**Reply by vhong8820** [reply]
Would there be a way to tell if the institution is running its own SEB server?

**Reply by nxvvvv** [reply]
To be honest, I have no idea.

**Reply by al3naziiy** [reply]
Hi I have a problem. When I connect to the university’s private network during the test, this error shows me. Knowing that it is on a network other than the test, it works correctly for me. What is the solution?![image](https://github.com/nxvvvv/safe-exam-browser-bypass/assets/130878356/0bbf1434-678f-4a58-bdb1-491d5fd0304b)

**Reply by nxvvvv** [reply]
Could you share the logs? The log files can be found under ``%LocalAppData%\SafeExamBrowser\Logs``.

**Comment by nxvvvv** [comment]
> Hi I have a problem. When I connect to the university’s private network during the test, this error shows me. Knowing that it is on a network other than the test, it works correctly for me. What is the solution?![image](https://private-user-images.githubusercontent.com/130878356/292777769-0bbf1434-678f-4a58-bdb1-491d5fd0304b.jpeg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDM3NTM5NjUsIm5iZiI6MTcwMzc1MzY2NSwicGF0aCI6Ii8xMzA4NzgzNTYvMjkyNzc3NzY5LTBiYmYxNDM0LTY3OGYtNGE1OC1iZGIxLTQ5MWQ1ZmQwMzA0Yi5qcGVnP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQUlXTkpZQVg0Q1NWRUg1M0ElMkYyMDIzMTIyOCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyMzEyMjhUMDg1NDI1WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MjkwZWJmNmQzNGIyOGU3NGEwMzY4NGVjM2FlNGM0NzlhNmUyM2JjN2Y2YWEzY2Y4ODZlMTVjMzk1ZGNiODIwYiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.thaYJA31f9lqQeAdQAo4pMvfhLHIKUAPA8UdDY6eZbs)

I can point to a probable cause, which is invalid TCP/IP Settings or windows firewall stopping the access of the modded SEB. Try running these commands in an admin terminal:
```netsh winsock reset```
```netsh int ip reset```
```ipconfig /release```
```ipconfig /renew```
```ipconfig /flushdns```

and then going try to turn off firewall just for the exam. [Check how to do that here](https://support.microsoft.com/en-us/windows/turn-microsoft-defender-firewall-on-or-off-ec0844f7-aebd-0583-67fe-601ecf5d774f)

**Reply by Nilanjan-B** [reply]
Hi brother, I've two questions basically. First, Do i need to run these commands in Host Machine or inside VM? Second, Is your readme.md method working on SEB 3.6.0 ? If not, then is it necessary to run the latest seb i.e., 3.6.0 for my upcoming exam or I can run your recommended SEB 3.5.0.544 for my exam? Thanks @nxvvvv for this repo work.

**Reply by nxvvvv** [reply]
It worked for a while. I'll update the patch and yea. I'd suggest you to use the 3.5 version since this is for that version

**Reply by Nilanjan-B** [reply]
As you're suggesting for using v3.5, **Is my university going to let me use this version or they will ask me to update it to the latest version?** Please clarify. By the way, I've just used your patch with 3.5 & SEB has opened without any issue. **Thanks for your time & effort.
.**

**Reply by nxvvvv** [reply]
> **Is my university going to let me use this version or they will ask me to update it to the latest version?** 

To be honest, No. As I remember, they will ask you to update if you have the 2.4 version so yea

**Reply by Nilanjan-B** [reply]
I've been using v3.5 with your Patches. Can i give my upcoming exams without updating it to the latest stable release v3.6 or Will university mandatorily ask me to update it to v3.6 from my current installed v3.5?

**Reply by lathianamc** [reply]
It comes down to the policies your University has placed. In my case, they run their server and in the policies they have set the version to the latest stable release, meaning, anything below v3.6 won't work. If you try to run the SEB url they provide you for the exam with v3.5, once it runs it will fail with a mismatched required version (if they have placed a rule to use v3.6).

**Reply by nxvvvv** [reply, marked as answer]
I've updated the patches to 3.6.0.633. Do check it out.
